services/scanner/agentpool/poolagent/agent.go

```diff
@@ -219,46 +227,90 @@ func (agent *Agent) SetClient(agentClient clients.AgentClient) {
 // from request channels.
 func (agent *Agent) StartProcessing() {
 	agent.initWait.Add(1)
-	go agent.initialize()
 
+	go func() {
+		err := agent.initWorker(agent.ctx)
+		if err != nil {
+			log.WithError(err).Warn("failed to initialize bot")
+		}
+	}()
 	go agent.processTransactions()
 	go agent.processBlocks()
 	go agent.processCombinationAlerts()
 }
 
-func (agent *Agent) initialize() {
-	defer agent.initWait.Done()
+func (agent *Agent) initWorker(ctx context.Context) error {
```

Comment:
The `cancel()`s inside this method seems to be complicating the code. For simplicity's sake, maybe it makes sense to move the cancellable context initialization to the beginning of `agent.initialize()` like:
```go
func (agent *Agent) initialize(ctx context.Context) {
  initCtx, cancel := context.WithTimeout(ctx)
  defer cancel()
  // initialize the bot
}
```
That method seems to be the single place where we make the actual call using the client so we can see both the timeout and the call itself there.

Reply:
i think you're right, i was afraid to add a defer where we have an "endless" for loop, but it should be safe